src/WebJobs.Extensions.CosmosDB/Services/CosmosDBService.cs

```diff
@@ -15,15 +16,16 @@ internal sealed class CosmosDBService : ICosmosDBService, IDisposable
         private bool _isDisposed;
         private DocumentClient _client;
 
-        public CosmosDBService(string connectionString, ConnectionPolicy connectionPolicy)
+        public CosmosDBService(string connectionString, ConnectionPolicy connectionPolicy, bool useDefaultJsonSerialization)
         {
             CosmosDBConnectionString connection = new CosmosDBConnectionString(connectionString);
             if (connectionPolicy == null)
             {
                 connectionPolicy = new ConnectionPolicy();
             }
 
-            _client = new DocumentClient(connection.ServiceEndpoint, connection.AuthKey, connectionPolicy);
+            var serializerSettings = useDefaultJsonSerialization ? JsonConvert.DefaultSettings?.Invoke() : default(JsonSerializerSettings);
+            _client = new DocumentClient(connection.ServiceEndpoint, connection.AuthKey, connectionPolicy, null, serializerSettings);
```

Comment:
This is passing null to the 4th parameter, even when the serializer is not configured on the new attribute. Also, if the user sets the new flag but doesn't configure the required setting, we are silently setting it to the default, in which case, the user would expect a different behavior. Better to throw so the user fixes the missing configuration IMO

Why not do something like:

if (useDefaultJsonSerialization){
 if (JsonConvert.DefaultSettings == null){
throw new ArgumentNullException("If UseDefaultJsonSerialization is enabled, JsonConvert.DefaultSettings should be configured.");
}
 _client = new DocumentClient(connection.ServiceEndpoint, connection.AuthKey, connectionPolicy, null, JsonConvert.DefaultSettings.Invoke());
}
else{
_client = new DocumentClient(connection.ServiceEndpoint, connection.AuthKey, connectionPolicy);
}

Reply:
Done